specifications/api/swagger.yaml

```diff
@@ -2601,6 +2674,37 @@ components:
         passphrase: *lenientPassphrase
         address_index: *addressIndex
 
+    ApiValidityInterval: &ApiValidityInterval
+      type: object
+      required:
+        - invalid_before
+        - invalid_hereafter
```

Comment:
We would want the Haskell validity bound type to stay the same, but in JSON, the bounds should be optional, with a missing key meaning "unspecified".

Reply:
so we have at this moment : 
```
2693     ApiConstructTransactionData: &ApiConstructTransactionData                                                                                                                                                 
2694       type: object                                                                                                                                                                                            
2695       required:                                                                                                                                                                                               
2696         - passphrase                                                                                                                                                                                          
2697       properties:                                                                                                                                                                                             
2698         passphrase:                                                                                                                                                                                           
2699           <<: *lenientPassphrase                                                                                                                                                                              
2700           description: The wallet's master passphrase.                                                                                                                                                        
2701         payments: *ApiPaymentDestination                                                                                                                                                                      
2702         withdrawal: *transactionWithdrawalRequestSelf                                                                                                                                                         
2703         metadata: *transactionMetadata                                                                                                                                                                        
2704         mint: *transactionMint                                                                                                                                                                                
2705         delegations: *transactionDelegations                                                                                                                                                                  
2706         validity_interval: *ApiValidityInterval
```
so the idea was to sneak optionality by making `validity_interval` optional, but make `ApiValidityInterval` well defined. kinda the same as for `ApiPaymentDestination` - the same idea...Once again I do not have strong opinion about this, could be either way, but maybe it would be ok to be consistent..

Reply:
For the JSON side, we want an omitted field to mean unspecified bound.